app/frontend/src/lib/message-service.ts

```diff
@@ -1,25 +1,31 @@
 import Dexie from "dexie";
 import type { EntityTable } from "dexie";
+import type { User } from "@server/db/schema.sql";
 
 export interface Message {
 	id?: string;
 	content: string;
 	timestamp: number;
 	status: "sent" | "received";
+	userId: string;
+	username: string;
 }
 
 const messageDb = new Dexie("MessageDatabase") as Dexie & {
 	messages: EntityTable<Message, "id">;
 };
 
-messageDb.version(1).stores({
-	messages: "++id, content, timestamp, type",
+messageDb.version(4).stores({
+	messages: "++id, content, timestamp, status, userId, username",
 });
 
+// * MessageService singleton for managing messages
 class MessageService {
 	private static instance: MessageService;
-
-	private constructor() {}
+	private currentUser: User | null = null;
```

Comment:
no need for currentuser to be a member of the messageservice, the ID can be injected at the message object initialization